src/components/TodoList/TodoList.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { TodosContext } from '../TodosContext';
+import { TodoItem } from '../TodoItem';
+import { TodosFilter } from '../TodosFilter';
+import { filterTodos } from '../../filterTodos';
+
+export const TodoList = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const filter = useLocation().pathname;
+
+  const allCompleted = useMemo(() => (todos.length
+    === filterTodos(todos, '/completed').length), [todos]);
```

Comment:
```suggestion
  const allCompleted = useMemo(() => (
    todos.length === filterTodos(todos, '/completed').length
  ), [todos]);
```
я б ось так переніс, легше читати

Reply:
спасибо) сегодня все исправлю